### ISSUE TITLE
Empty blocks display no selection state

### ISSUE THREAD

**Issue by steobrien** [ianstormtaylor/slate#2275]
#### Do you want to request a _feature_ or report a _bug_?

Bug.

#### What's the current behavior?

When selecting, for empty blocks, no selection state is visible:
![image](https://user-images.githubusercontent.com/1694410/47055547-e5240a00-d185-11e8-8974-7a58ba04de6e.png)

#### What's the expected behavior?

The empty blocks should have a minimal selection state. This improves usability when making selections with keyboard or mouse.

Some examples from other contexts:

- Plain textarea:
![image](https://user-images.githubusercontent.com/1694410/47055580-0be24080-d186-11e8-8e87-88de50fc6d2f.png)
- Draft.js:
![image](https://user-images.githubusercontent.com/1694410/47055594-2e745980-d186-11e8-8ebf-0e892595092c.png)
- Quill:
![image](https://user-images.githubusercontent.com/1694410/47055630-611e5200-d186-11e8-8b05-aaabdee12939.png)
- Trix:
![image](https://user-images.githubusercontent.com/1694410/47055640-709d9b00-d186-11e8-810b-28f80c4b5943.png)
- ProseMirror:
![image](https://user-images.githubusercontent.com/1694410/47055649-7c895d00-d186-11e8-94e9-db20af559517.png)

**Comment by ericedem** [comment]
Changing the character rendered in empty blocks from a zero width char to a normal space fixes this issue. And copy pasting still works as far as I can tell:

https://github.com/ianstormtaylor/slate/blob/master/packages/slate-react/src/components/leaf.js#L145

Anyone see any drawbacks to this?

**Comment by steobrien** [comment]
@ericedem I think the issue then will be that clicking on an empty block can momentarily show the cursor in the wrong position:
<img src="https://user-images.githubusercontent.com/1694410/47218132-c89bf380-d378-11e8-81f3-e4b8db3bef3b.gif" width="200" />

**Comment by bryanph** [comment]
What browser do you use? In my case (firefox, linux), the selection is never highlighted like you mentioned for empty blocks.

**Comment by ericedem** [comment]
@bryanph I think @steobrien is just demonstrating what happens when you render a normal space character like I suggested instead of the normal zero width character.

As I was experimenting with some other styling solutions, I noticed that Chrome and Firefox (on mac) don't show the selection for empty blocks. Safari is very different, and expands the selection across the entire component.

I also noticed a crazy behavior after playing around with it, Chrome started showing the selection on the zero width character 😖. Seemingly for no reason.
